presto-spi/src/main/java/com/facebook/presto/spi/block/RowBlockBuilder.java

```diff
@@ -45,15 +45,14 @@
 
     private boolean currentEntryOpened;
 
-    public RowBlockBuilder(List<Type> fieldTypes, BlockBuilderStatus blockBuilderStatus, int expectedEntries)
+    public static RowBlockBuilder createRowBlockBuilder(List<Type> fieldTypes, BlockBuilderStatus blockBuilderStatus, int expectedEntries)
     {
-        this(
-                blockBuilderStatus,
-                fieldTypes.stream()
-                        .map(type -> type.createBlockBuilder(blockBuilderStatus, expectedEntries))
-                        .toArray(BlockBuilder[]::new),
```

Comment:
You could have retained the constructor and covert the Stream-based code into plain static method